lib/brakeman/checks/check_unscoped_find.rb

```diff
@@ -52,12 +52,16 @@ def process_result result
       :cwe_id => [285]
   end
 
-  def optional_belongs_to? exp
+  def self_referential_association? exp, name
     return false unless exp.is_a? Array
 
     exp.each do |e|
       if hash? e and true? hash_access(e, :optional)
-        return true
+        hash_iterate e do |key, value|
```

Comment:
Is `class_name` going to show up more than once as a key? Couldn't this also use `hash_access(:class_name)` instead of iterating?